src/diff/index.js

```diff
@@ -215,6 +224,12 @@ export function diff(
 
 		if ((tmp = options.diffed)) tmp(newVNode);
 	} catch (e) {
+		if (isHydrating) {
```

Comment:
This is literally the only thing I did in this PR.
Rest everything is @developit's code. 
Would love to send it to those PRs instead if that is preferred.